### PR TITLE
Adding execution plan component to model view. 

### PR DESCRIPTION
**Pull request by aasimkhan30**
Test:
image

Usage example: 

```
this.executionPlan = view.modelBuilder.executionPlan().withProps({
				data: {
					graphFileContent: file xml
					planIndexInFile: 0,
					graphFileType: 'sqlplan'
				}
			}).component();
```

Updating example:
Using graphs
```
this.executionPlan.data = <azdata.executionPlan.ExecutionPlanGraph[]>data
```

Using graph files
```
this.executionPlan.data = {
					graphFileContent: 'file xml'
					planIndexInFile: 0,
					graphFileType: 'sqlplan'
				}
```

### CODE REVIEW

File: src/sql/workbench/browser/modelComponents/executionPlan.component.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	//no-op
+	override setLayout(layout: any): void {
+	}
+
+	@Input() descriptor: IComponentDescriptor;
+	@Input() modelStore: IModelStore;
+
+	@ViewChild('executionPlanContainer') private _container: ElementRef;
+
+	private _data: azdata.executionPlan.ExecutionPlanGraphInfo | azdata.executionPlan.ExecutionPlanGraph[];
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modelComponents/executionPlan.component.ts]: `| undefined`

File: src/sql/workbench/browser/modelComponents/executionPlan.component.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		super.setProperties(properties);
+		if (properties.data) {
+
+
+			if (equals(this._data, properties.data) === false) {
+				this._data = properties.data;
+				this.clearExecutionPlan();
+				this.loadData(this._data);
+			}
+		} else {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modelComponents/executionPlan.component.ts:71]: remove?

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanFileView.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ export class ExecutionPlanFileView extends Disposable {
 	private _planCache: Map<string, azdata.executionPlan.ExecutionPlanGraph[]> = new Map();
 
 	constructor(
-		private _queryResultsView: QueryResultsView,
+		private _queryResultsView: QueryResultsView | undefined,
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanFileView.ts:31]: Did you make sure everything below is handling undefined correctly?

**Review by Charles-Gagnon — APPROVED** [review]

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanFileView.ts (hunk 3 above)
Comment (aasimkhan30) [reply src/sql/workbench/contrib/executionPlan/browser/executionPlanFileView.ts:31]: Yup, I have checked that case

File: src/sql/workbench/browser/modelComponents/executionPlan.component.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	}
+
+	@Input() descriptor: IComponentDescriptor;
+	@Input() modelStore: IModelStore;
+
+	@ViewChild('executionPlanContainer') private _container: ElementRef;
+
+	private _data: azdata.ExecutionPlanData | undefined;
+	private _executionPlanFileView: ExecutionPlanFileView;
+	s
```
Comment (kisantia) [inline src/sql/workbench/browser/modelComponents/executionPlan.component.ts:37]: remove `s` on line 37?

File: src/sql/workbench/browser/modelComponents/executionPlan.component.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+		this.baseDestroy();
+		this._executionPlanFileView.dispose();
+		this._data = undefined;
+	}
+
+	public override setProperties(properties: { [key: string]: any; }): void {
+		super.setProperties(properties);
+		if (properties.data) {
+
+
```
Comment (kisantia) [inline src/sql/workbench/browser/modelComponents/executionPlan.component.ts:65]: nit: extra blank line

**Review by kisantia — APPROVED** [review]

**Review by lewis-sanchez — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

**Comment by Charles-Gagnon** [comment]
@aasimkhan30 You merged in the distro PR early so this is blocking further updates there unless we roll that back. Are you waiting on anything or can we get this in?

For future reference - you should never merge the distro PR before completing the corresponding one in ADS

**Comment by Charles-Gagnon** [comment]
See https://github.com/microsoft/azuredatastudio-release/wiki/Updating-the-Release-Repo#how-to-resolve-merge-conflict-issues for the process to follow

**Pull request merged**